pkg/assembler/backends/testing/hashEqual.go

```diff
@@ -17,113 +17,205 @@ package testing
 
 import (
 	"context"
-	"reflect"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
 	"strings"
 
-	"github.com/guacsec/guac/pkg/assembler/graphql/model"
 	"github.com/vektah/gqlparser/v2/gqlerror"
+	"golang.org/x/exp/slices"
+
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
 )
 
-func registerAllHashEqual(client *demoClient) {
+// Internal hashEqual
 
-	// strings.ToLower(string(checksum.Algorithm)) + ":" + checksum.Value
-	client.registerHashEqual([]*model.Artifact{client.artifacts[0], client.artifacts[1], client.artifacts[2]}, "different algorithm for the same artifact", "testing backend", "testing backend")
-	client.registerHashEqual([]*model.Artifact{{Digest: "5a787865sd676dacb0142afa0b83029cd7befd9", Algorithm: "sha1"},
-		{Digest: "89bb0da1891646e58eb3e6ed24f3a6fc3c8eb5a0d44824cba581dfa34a0450cf", Algorithm: "sha256"}}, "these two are the same", "testing backend", "testing backend")
+type hashEqualList []*hashEqualStruct
+type hashEqualStruct struct {
+	id            uint32
+	artifacts     []uint32
+	justification string
+	origin        string
+	collector     string
 }
 
-// Ingest HashEqual
+func (n *hashEqualStruct) getID() uint32 { return n.id }
 
-func (c *demoClient) registerHashEqual(artifacts []*model.Artifact, justification, origin, collector string) *model.HashEqual {
+func registerAllHashEqual(client *demoClient) {
```

Comment:
can remove this as we have test in cmd/graphql_playground/cmd/ingest. The only ones we dont have are the objects so like artifact would need it.

Reply:
yeah that is fine. We can transfer it over to the unit tests.